core/src/test/resources/saffron.properties

```diff
@@ -0,0 +1,17 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to you under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+calcite.planner.topdown.opt=true
```

Comment:
Tests run with enabled top-down optimization. But by default this optimization is turned off. So we don't test the default optimizer behavior. I think it can lead to regressions in the legacy optimization workflow. May be we can test both cases somehow?

Reply:
It does test the default behavior. It doesn't change the rule apply behavior, just disabled the use of AbstractConverter, which was never enabled nor tested until CALCITE-2970. When we implement top-down rule apply, we might need to test both.

Reply:
So the shipping configuration is false by default (CalciteSystemProperties), while the UT is set to true? I believe our UT should run on default shipping configs. In such case, we would need new test suite (or test matrix) with this config set to true.

Reply:
Here is what it looks like. We ship 2 products: A and B, B is turned off by default. But we test both A and B in test environment.